control/runner.go

```diff
@@ -66,11 +67,17 @@ type executablePlugin interface {
 	WaitForResponse(time.Duration) (*plugin.Response, error)
 }
 
+type scheduler interface {
```

Comment:
I'm not sure if `scheduler` is not misleading here. Maybe would be good to think of another, more suitable name for this interface